csrc/device_lower/analysis/thread_predicate.cpp

```diff
@@ -494,6 +504,192 @@ class RedundantUseAnalysis : BackwardVisitor {
 
 } // namespace
 
+namespace {
+// This class removes the redundant write to gmem when an output tensor has a
+// leaf domain merged from concretized broadcast root domain and parallelized by
+// thread/block id. issue https://github.com/csarofeen/pytorch/issues/2125
+class ConcretizedBroadcastRedundantWriteRemover {
+ public:
+  // interface to run the check
+  ConcretizedBroadcastRedundantWriteRemover(const TensorView* out_tv)
+      : tv_(out_tv), root_domain_(out_tv->getRootDomain()) {
```

Comment:
I think this should be `out_tv->getMaybeRfactorDomain()`

Reply:
corrected.